Add support for svg inlining

josbeir opened the pull request:
This adds support for the `inline` option per set allowing to compact svg's making your document less bloated.

josbeir committed:
Add support for svg inlining

dereuromark commented on src/View/Icon/SvgRenderTrait.php:
```diff
@@ -70,6 +70,11 @@ protected function renderSvg(string $icon, array $attributes = []): HtmlStringab
 		}
 
+		// Apply inlining if enabled
+		if ($this->config['inline'] ?? false) {
```
We could default to `!Configure::read(''debug')` for having this auto done for prod. Can be manually set to true/false still.

josbeir replied:
added!

josbeir commented on tests/TestCase/View/Icon/LucideIconTest.php:
```diff
@@ -203,4 +203,228 @@ public function testRenderSvgFromJsonMapThrowsExceptionWhenIconNotFound(): void
+	public function testRenderSvgFromJsonMapWithInlining(): void {
+		$jsonFile = TMP . 'tests' . DS . 'lucide-icons-inline.json';
+		$jsonContent = json_encode([
+			'clock' => '<!-- Comment in content -->
+<path d="M12 6v6l4 2"/>  
+<circle cx="12" cy="12" r="10"/>
+<!-- Another comment -->',
```
need for white space testing, could use heredoc too but seems unnecessary

josbeir committed:
Added `Configure('debug')` checking support and auto inline when `false`
Added the `inline` argument to all icon classes for improved readability
Updated docs

josbeir committed:
Improve inline regex functionality

josbeir committed:
phpstan fix
